mathlibtools/lib.py

```diff
@@ -478,8 +478,11 @@ def get_cache(self, rev: Optional[str] = None, force: bool = False) -> None:
         """
         if not self.repo:
             raise LeanProjectError('This project has no git repository.')
-        if self.is_dirty and not force:
-            raise LeanDirtyRepo('Cannot get cache for a dirty repository.')
+        if self.is_dirty and rev is None and not force:
+            # if the user passed `rev`, they deliberately want a cache that does
+            # not match their local changes anyway.
+            raise LeanDirtyRepo('Getting the cache may result in a '
+                                'time-consuming rebuild.')
```

Comment:
I don't understand this error message. It looks like a warning to the user and will still stop everything.

Reply:
Right - the point is that the user is about to blow away any local olean files they might have from modifications, and once they do that the only way to get them back is to rebuild (at least, I think that's how this works).

So what I'm trying to convey is "if you do this by accident you may end up losing time, pass `--force` to proceed".

I see now that this code isn't quite right either. There are two different types of caches:

* `olean`-only caches, like those uploaded to azure for mathlib
* `lean` + `olean` caches, like those made with `mk-cache`

In the first case, `--force` is really just about being willing to wait a bit longer. In the second case, `--force` can actually destroy changes in your working tree.